public/app/features/correlations/utils.ts

```diff
@@ -43,7 +44,7 @@ export const attachCorrelationsToDataFrames = (
     }
 
     const sourceCorrelations = correlations.filter((correlation) => correlation.source.uid === dataSourceUid);
-    decorateDataFrameWithInternalDataLinks(dataFrame, sourceCorrelations);
+    decorateDataFrameWithInternalDataLinks(dataFrame, fixLokiDataplaneFields(sourceCorrelations, dataFrame));
```

Comment:
This would fix it only in runtime. The underlying correlation would still stay the same in the database, wouldn't it? I wonder if in long-run it'd be better to allow data sources to (somehow 🤷 ) migrate correlations 🤔